db/migration/src/m20240226_000001_create_channel.rs

```diff
@@ -0,0 +1,96 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Channel::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(Channel::Id)
+                            .integer()
+                            .not_null()
+                            .auto_increment()
+                            .primary_key(),
+                    )
+                    .col(
+                        ColumnDef::new(Channel::ChannelId)
+                            .string_len(64)
+                            .not_null()
+                            .unique_key(),
+                    )
+                    .col(ColumnDef::new(Channel::Source).string_len(40).not_null())
+                    .col(ColumnDef::new(Channel::Destination).string_len(40).not_null())
+                    .col(ColumnDef::new(Channel::Balance).string_len(50).not_null())
+                    .col(ColumnDef::new(Channel::Status).tiny_integer().not_null())
+                    .col(
+                        ColumnDef::new(Channel::Epoch)
+                            .integer()
+                            .unsigned()
+                            .not_null()
+                            .default(1),
+                    )
+                    .col(
+                        ColumnDef::new(Channel::TicketIndex)
+                            .integer()
+                            .unsigned()
+                            .not_null()
+                            .default(0),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        manager
+            .create_index(
+                Index::create()
+                    .if_not_exists()
+                    .name("idx_channel_source")
+                    .table(Channel::Table)
+                    .col(Channel::Source)
+                    .to_owned(),
+            )
+            .await?;
+
+        manager
```

Comment:
Possibly an index covering source and destination would make sense here too?

Reply:
Can be added yes.